packages/amplify-graphql-transformer-core/src/transformation/sync-utils.ts

```diff
@@ -90,7 +90,7 @@ export function syncDataSourceConfig(): DeltaSyncConfig {
 export function validateResolverConfigForType(ctx: TransformerSchemaVisitStepContextProvider, typeName: string): void {
   const resolverConfig = ctx.getResolverConfig<ResolverConfig>();
   const typeResolverConfig = resolverConfig?.models?.[typeName];
-  if (!typeResolverConfig || !typeResolverConfig.ConflictDetection || !typeResolverConfig.ConflictHandler) {
+  if (typeResolverConfig && !typeResolverConfig.ConflictDetection && !typeResolverConfig.ConflictHandler) {
```

Comment:
This will not warn the user if they only have a project sync configuration but not for the model

Reply:
Ah correct, though this warning should show up if the typeResolverConfig key exists yet the values are incomplete. Maybe something like this?
```suggestion
  if (typeResolverConfig && (!typeResolverConfig.ConflictDetection || !typeResolverConfig.ConflictHandler)) {
```

Reply:
its possible they only have `resolverConfig?.project;` but not `resolverConfig?.models?.[typeName];` so in this case `typeResolverConfig` will be undefined